src/make_watertight/Cleanup.cpp

```diff
@@ -0,0 +1,390 @@
+    result = MBI()->merge_entities(vert0, vert1, false, true);
+    assert(moab::MB_SUCCESS == result);
+    vert1 = vert0;
+  }
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Cleanup::delete_small_edges(const moab::Range& surfaces, const double FACET_TOL) {
+  // PROBLEM: THIS IS INVALID BECAUSE TRIS CAN HAVE LONG EDGES BUT
+  // SMALL AREA. All three pts are in a line. This is the nature of
+  // faceting vs. meshing.
+  /* Remove small triangles by removing edges that are too small.
+  Remove small edges by merging their endpoints together, creating
+  degenerate triangles. Delete the degenerate triangles. */
+  moab::ErrorCode result;
+  for (moab::Range::const_iterator i = surfaces.begin(); i != surfaces.end(); i++) {
+    std::cout << "surf_id=" << gen->geom_id_by_handle(*i) << std::endl;
+
+    // get all tris
+    moab::Range tris;
+    result = MBI()->get_entities_by_type(*i, moab::MBTRI, tris);
+    assert(moab::MB_SUCCESS == result);
+
+    // Check to ensure there area no degenerate tris
+    for (moab::Range::iterator j = tris.begin(); j != tris.end(); j++) {
+      // get endpts
+      const moab::EntityHandle* endpts;
+      int n_verts;
+      result = MBI()->get_connectivity(*j, endpts, n_verts);
+      assert(moab::MB_SUCCESS == result);
+      assert(3 == n_verts);
+      assert(endpts[0] != endpts[1] && endpts[1] != endpts[2]);
+    }
+
+
+    // get the skin first, because my find_skin does not check before creating edges.
+    moab::Range skin_edges;
+    //result = gen->find_skin( tris, 1, skin_edges, false );
+    moab::Skinner tool(MBI());
+    result = tool.find_skin(0, tris, 1, skin_edges, false);
+    assert(moab::MB_SUCCESS == result);
+
+    // create the edges
+    moab::Range edges;
+    result = MBI()->get_adjacencies(tris, 1, true, edges, moab::Interface::UNION);
+    if (moab::MB_SUCCESS != result) {
+      std::cout << "result=" << result << std::endl;
+    }
+    assert(moab::MB_SUCCESS == result);
+
+    // get the internal edges
+    moab::Range internal_edges = subtract(edges, skin_edges);
+
+    for (moab::Range::iterator j = internal_edges.begin(); j != internal_edges.end(); j++) {
+      int n_internal_edges = internal_edges.size();
+      std::cout << "edge=" << *j << std::endl;
+      MBI()->list_entity(*j);
+      assert(moab::MB_SUCCESS == result);
+
+      // get endpts
+      const moab::EntityHandle* endpts;
+      int n_verts;
+      result = MBI()->get_connectivity(*j, endpts, n_verts);
+      assert(moab::MB_SUCCESS == result);
+      assert(2 == n_verts);
+
+      // does another edge exist w the same endpts? Why would it?
+      moab::Range duplicate_edges;
+      result = MBI()->get_adjacencies(endpts, 2, 1, true, duplicate_edges);
+      assert(moab::MB_SUCCESS == result);
+      if (1 < duplicate_edges.size())
+        MBI()->list_entities(duplicate_edges);
+      assert(1 == duplicate_edges.size());
+
+      // if the edge length is less than MERGE_TOL do nothing
```

Comment:
Comment looks out of date? Looks like the comparison is for greater than `FACET_TOL`